### ISSUE TITLE
"Failed to obtain XInput device capabilities" when streaming to tablet using Steam Link.

### ISSUE THREAD

**Issue by Belgerum** [renpy/renpy#2067]
When using Steam Link to stream my newest Ren'Py game to a tablet (an Apple iPad, in my case), my game will often raise an exception on launch with the following traceback. Ignoring the exception causes the game to run normally afterward. It only seems to be affecting my most recent game, so this error is likely rooted in a Ren'Py update sometime since August.

```
I'm sorry, but an uncaught exception occurred.

While running game code:
  File "renpy/common/00start.rpy", line 207, in script call
    call _gl_test
  File "renpy/common/00gltest.rpy", line 445, in script
    $ __gl_test()
  File "renpy/common/00gltest.rpy", line 445, in <module>
    $ __gl_test()
  File "renpy/common/00gltest.rpy", line 359, in _m1_00gltest__gl_test
    _gl_performance_test()
  File "renpy/common/00gltest.rpy", line 409, in _gl_performance_test
    result = ui.interact(suppress_overlay=True, suppress_underlay=True)
error: Failed to obtain XInput device capabilities. Device disconnected?

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "renpy/common/00start.rpy", line 207, in script call
    call _gl_test
  File "renpy/common/00gltest.rpy", line 445, in script
    $ __gl_test()
  File "C:\Program Files (x86)\Steam\steamapps\common\Steamy Sextet\renpy\ast.py", line 914, in execute
    renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
  File "C:\Program Files (x86)\Steam\steamapps\common\Steamy Sextet\renpy\python.py", line 2028, in py_exec_bytecode
    exec bytecode in globals, locals
  File "renpy/common/00gltest.rpy", line 445, in <module>
    $ __gl_test()
  File "renpy/common/00gltest.rpy", line 359, in _m1_00gltest__gl_test
    _gl_performance_test()
  File "renpy/common/00gltest.rpy", line 409, in _gl_performance_test
    result = ui.interact(suppress_overlay=True, suppress_underlay=True)
  File "C:\Program Files (x86)\Steam\steamapps\common\Steamy Sextet\renpy\ui.py", line 297, in interact
    rv = renpy.game.interface.interact(roll_forward=roll_forward, **kwargs)
  File "C:\Program Files (x86)\Steam\steamapps\common\Steamy Sextet\renpy\display\core.py", line 2702, in interact
    repeat, rv = self.interact_core(preloads=preloads, trans_pause=trans_pause, **kwargs)
  File "C:\Program Files (x86)\Steam\steamapps\common\Steamy Sextet\renpy\display\core.py", line 3500, in interact_core
    ev = renpy.display.controller.event(ev)
  File "C:\Program Files (x86)\Steam\steamapps\common\Steamy Sextet\renpy\display\controller.py", line 162, in event
    start(ev.which)
  File "C:\Program Files (x86)\Steam\steamapps\common\Steamy Sextet\renpy\display\controller.py", line 150, in start
    c.init()
  File "src/pygame_sdl2/controller.pyx", line 160, in pygame_sdl2.controller.Controller.init
error: Failed to obtain XInput device capabilities. Device disconnected?

Windows-8-6.2.9200
Ren'Py 7.3.5.606
Steamy Sextet Demo
Thu Nov 21 23:55:59 2019
```

**Comment by Belgerum** [comment]
I've been unable to reproduce this error again after the game formally launched to the public, so I suppose I can assume that this issue only occurs for unreleased games. Thus, not a huge issue for anyone except developers and beta testers, but still an issue nonetheless.

**Comment by renpytom** [comment]
Chances are this was fixed by the changes in the 7.4 series. If not, please re-open this issue.
